### ISSUE TITLE
`non_camel_case_name` / `non_snake_case_name` / `non_upper_case_globals` don't fire for `use x as y`

### ISSUE THREAD

**Issue by LunarLambda** [rust-lang/rust#126902]
### Code

```Rust
// doesn't warn
use Option as option;

// doesn't warn
use drop as Drop;

const FOO: () = ();
// doesn't warn
use FOO as foo;

// does warn
enum yip {}

// does warn
fn Woof() {}

// does warn
const mya: () = ();
```

### Current output

```Shell
unused import warnings, but not name warnings on the imports

warning: unused import: `Option as option`
 --> src/lib.rs:2:5
  |
2 | use Option as option;
  |     ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `drop as Drop`
 --> src/lib.rs:5:5
  |
5 | use drop as Drop;
  |     ^^^^^^^^^^^^

warning: unused import: `FOO as foo`
 --> src/lib.rs:9:5
  |
9 | use FOO as foo;
  |     ^^^^^^^^^^

warning: type `yip` should have an upper camel case name
  --> src/lib.rs:12:6
   |
12 | enum yip {}
   |      ^^^ help: convert the identifier to upper camel case (notice the capitalization): `Yip`
   |
   = note: `#[warn(non_camel_case_types)]` on by default

warning: constant `FOO` is never used
 --> src/lib.rs:7:7
  |
7 | const FOO: () = ();
  |       ^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: enum `yip` is never used
  --> src/lib.rs:12:6
   |
12 | enum yip {}
   |      ^^^

warning: function `Woof` is never used
  --> src/lib.rs:15:4
   |
15 | fn Woof() {}
   |    ^^^^

warning: constant `mya` is never used
  --> src/lib.rs:18:7
   |
18 | const mya: () = ();
   |       ^^^

warning: function `Woof` should have a snake case name
  --> src/lib.rs:15:4
   |
15 | fn Woof() {}
   |    ^^^^ help: convert the identifier to snake case (notice the capitalization): `woof`
   |
   = note: `#[warn(non_snake_case)]` on by default

warning: constant `mya` should have an upper case name
  --> src/lib.rs:18:7
   |
18 | const mya: () = ();
   |       ^^^ help: convert the identifier to upper case: `MYA`
   |
   = note: `#[warn(non_upper_case_globals)]` on by default
```

### Desired output

```Shell
import renames and re-exports should still be subject to the naming lints.
```

### Rust Version

```Shell
1.79.0 stable, tested on play.rust-lang.org
```
